hs_core/views/__init__.py

```diff
@@ -568,6 +568,7 @@ def add_metadata_element(request, shortkey, element_name, *args, **kwargs):
             is_add_success = True
             res.update_public_and_discoverable()
             resource_modified(res, request.user, overwrite_bag=False)
+            res.update_cached_metadata_field('subject')
```

Comment:
Is this necessary because subjects is a property on the resource metadata model and does not use the AbstractMetadataElement model?

I guess you could listen to an event for saving the metadata model and look for changes to subject but this would be more efficient. Are you confident this will capture all updates to subjects? Is the subject cache updated when content type metadata extraction adds subject keywords?